pkgs/ok_http/android/src/main/kotlin/com/example/ok_http/RedirectInterceptor.kt

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// To cause a request failure [with a suitable message] due to too many redirects,
+// we need to throw an IOException. This cannot be done using Dart JNI bindings,
+// which lead to a deadlock and eventually a `java.net.SocketTimeoutException`.
+// https://github.com/dart-lang/native/issues/561
```

Comment:
Could you add a note to that issue explaining how you had to work around this issue?

Reply:
Done ✅ 
Here is the link - https://github.com/dart-lang/native/issues/561#issuecomment-2163696640

Let me know if there's something more you'd like me to add to it.